CONTRIBUTING.adoc

```diff
@@ -1,232 +1,140 @@
+By participating you are expected to uphold this code.
+Please report unacceptable behavior to spring-code-of-conduct@pivotal.io.
+
+[[how-to-contribute]]
+== How to Contribute
+
+[[ask-questions]]
+=== Ask Questions
+
+If you have a question, check Stack Overflow using
+https://stackoverflow.com/questions/tagged/spring-security+or+spring-ldap+or+spring-authorization-server+or+spring-session?tab=Newest[this list of tags].
+Find an existing discussion, or start a new one if necessary.
+
+If you believe there is an issue, search through https://github.com/spring-projects/spring-security/issues[existing issues] trying a  few different ways to find discussions, past or current, that are related to the issue.
+Reading those discussions helps you to learn about the issue, and helps us to make a decision.
+
+[[find-an-issue]]
+=== Find an Existing Issue
+
+There are many issues in Spring Security with the labels https://github.com/spring-projects/spring-security/issues?q=is%3Aissue+is%3Aopen+label%3A%22status%3A+ideal-for-contribution%22[`ideal-for-contribution`] or https://github.com/spring-projects/spring-security/issues?q=is%3Aissue+is%3Aopen+label%3A%22status%3A+first-timers-only%22[`first-timers-only`] that are a great way to contribute to a discussion or <<submit-a-pull-request,to a PR>>.
+You can volunteer by commenting on these tickets, and we will assign them to you.
+
+[[create-an-issue]]
+=== Create an Issue
+
+Reporting an issue or making a feature request is a great way to contribute.
+Your feedback and the conversations that result from it provide a continuous flow of ideas.
+However, before creating a ticket, please take the time to <<ask-questions,ask and research>> first.
+
+If you create an issue after a discussion on Stack Overflow, please provide a description in the issue instead of simply referring to Stack Overflow.
+The issue tracker is an important place of record for design discussions and should be self-sufficient.
+
+Once you're ready, create an issue on https://github.com/spring-projects/spring-framework/issues[GitHub].
+
+Many issues are caused by subtle behavior, typos, and unintended configuration.
+Creating a https://stackoverflow.com/help/minimal-reproducible-example[Minimal Reproducible Example] (starting with https://start.spring.io for example) of the problem helps the team quickly triage your issue and get to the core of the problem.
+
+We love contributors, and we may ask you to <<submit-a-pull-request,submit a PR with a fix>>.
+
+[[issue-lifecycle]]
+=== Issue Lifecycle
+
+When an issue is first created, it is flagged `waiting-for-triage` waiting for a team member to triage it.
+Once the issue has been reviewed, the team may ask for further information if needed, and based on the findings, the issue is either assigned a target branch (or no branch if a feature) or is closed with a specific status.
+The target branch is the earliest supported branch where the change will be applied.
```

Comment:
I think it is important to mention where the contributor can find the supported branches